src/explore-education-statistics-common/src/components/form/rhf/__tests__/RHFForm.test.tsx

```diff
@@ -0,0 +1,491 @@
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    // Stop the onSubmit from throwing error
+    onSubmit.mockImplementation();
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Something went wrong'),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test('removes submit error when form is reset', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: 'Firstname',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ reset }) => {
+          return (
+            <RHFForm
+              id="test-form"
+              showSubmitError
+              onSubmit={() => {
+                throw new Error('Something went wrong');
+              }}
+            >
+              The form
+              <button type="submit">Submit</button>
+              <button
+                type="button"
+                onClick={() => {
+                  reset();
+                }}
+              >
+                Reset form
+              </button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    fireEvent.click(screen.getByText('Reset form'));
+
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Something went wrong'),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test('removes submit error when form values are changed', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: 'Firstname',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm
+              id="test-form"
+              showSubmitError
+              onSubmit={() => {
+                throw new Error('Something went wrong');
+              }}
+            >
+              <label htmlFor="firstName">Firstname</label>
+              <input
+                type="text"
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    fireEvent.change(screen.getByLabelText('Firstname'), {
```

Comment:
Could use a `userEvent.type` to change this field? If we can, we have another `fireEvent.change` we could also get rid of further down (and we can get rid of `fireEvent` as an import).